ts/Extensions/Annotations/Types/FibonacciTimeZones.ts

```diff
@@ -0,0 +1,306 @@
+
+    It uses x coordinate to create two mock points on the same x. Then,
+    it uses some logic from InfinityLine to find equation of the line passing
+    through our two points and, using that equation, if finds and returns
+    the coordinates of where the line intersects the plot area edges.
+
+    This is being done for each fibonacci time zone line.
+
+            this point here is found
+     |---------*--------------------------------------------------------|
+     |                                                                  |
+     |                                                                  |
+     |                                                                  |
+     |                                                                  |
+     |         *   copy of the primary point                            |
+     |                                                                  |
+     |         *   primary point (e.g. the one given in options)        |
+     |                                                                  |
+     |---------*--------------------------------------------------------|
+            and this point here is found
+
+    */
+    private static edgePoint(
+        startIndex: number,
+        endIndex: number,
+        fibonacciIndex: number
+    ): Function {
+        return function (target: any): PositionObject {
+            const annotation = target.annotation;
+            let points = annotation.points;
+            // Offset between the first and the second line
+            const deltaX = points.length > 1 ? points[1].x - points[0].x : 0,
+                // firstLine.x + fibb * offset
+                x = points[0].x + fibonacciIndex * deltaX;
+
+            // We need 2 mock points with the same x coordinate, different y
+            points = [
+                new MockPoint(
+                    annotation.chart,
+                    points[0].target,
+                    {
+                        x: x,
+                        y: 0,
+                        xAxis: points[0].options.xAxis,
+                        yAxis: points[0].options.yAxis
+                    }
+                ),
+                new MockPoint(
+                    annotation.chart,
+                    points[0].target,
+                    {
+                        x: x,
+                        y: 1,
+                        xAxis: points[0].options.xAxis,
+                        yAxis: points[0].options.yAxis
+                    }
+                )
+            ];
+
+            return InfinityLine.findEdgePoint(
+                points[startIndex],
+                points[endIndex]
+            );
+        };
+    }
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public addShapes(): void {
+        const numberOfLines = 11;
+        let fibb = 1,
+            nextFibb = 1;
+
+        for (let i = 0; i < numberOfLines; i++) {
```

Comment:
No hard opinion here just a proposal to consider ;)

What if you start that loop from `1` and handle the first point outside that loop that you could avoid the `correctedFibb`

Reply:
I considered few approaches here and I think this one is short and easy. Maybe there is another one, but as long as it's not simpler/shorter, I leave it as it is now.